var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -683,16 +683,16 @@ def cmake_args(self):
                 # hence the test to see if the version starts with "flang".
                 targets.append('CppBackend')
 
-            if 'x86' in spec.architecture.target.lower():
+            if 'x86' in spec.target.family:
                 targets.append('X86')
-            elif 'arm' in spec.architecture.target.lower():
+            elif 'arm' in spec.target.family:
                 targets.append('ARM')
-            elif 'aarch64' in spec.architecture.target.lower():
+            elif 'aarch64' in spec.target.family:
                 targets.append('AArch64')
-            elif 'sparc' in spec.architecture.target.lower():
+            elif 'sparc' in spec.target.family:
```

Comment:
```suggestion
            elif spec.target.family == 'sparc' or spec.target.family == 'sparc64':
```
Probably this will be a niche case, but according to [this list of generic architectures](https://stackoverflow.com/questions/45125516/possible-values-for-uname-m) it should cover previous cases.